test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -22,63 +22,98 @@
 import io.pravega.client.tables.impl.KeyValueTableFactoryImpl;
 import io.pravega.client.tables.impl.KeyValueTableTestBase;
 import io.pravega.common.util.ByteArraySegment;
-import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+
 import io.pravega.segmentstore.contracts.StreamSegmentStore;
 import io.pravega.segmentstore.contracts.tables.TableStore;
 import io.pravega.segmentstore.server.host.handler.PravegaConnectionListener;
 import io.pravega.segmentstore.server.store.ServiceBuilder;
 import io.pravega.segmentstore.server.store.ServiceBuilderConfig;
-import io.pravega.test.common.AssertExtensions;
 import io.pravega.test.common.TestUtils;
-import java.time.Duration;
 import java.util.Collections;
+import java.time.Duration;
+import io.pravega.test.common.TestingServerStarter;
+import io.pravega.test.integration.demo.ControllerWrapper;
+import lombok.extern.slf4j.Slf4j;
 import lombok.val;
+import org.apache.curator.test.TestingServer;
 import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Ignore;
 import org.junit.Test;
 
-import static org.mockito.Mockito.mock;
-
 /**
  * Integration test for {@link KeyValueTable}s using real Segment Store and connection.
  * The only simulated component is the {@link Controller} which is provided via the {@link MockController}.
+ * import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+ * //import io.pravega.test.common.AssertExtensions;
  */
+@Slf4j
 public class KeyValueTableTest extends KeyValueTableTestBase {
     private static final String ENDPOINT = "localhost";
     private static final String SCOPE = "Scope";
     private static final KeyValueTableConfiguration DEFAULT_CONFIG = KeyValueTableConfiguration.builder().partitionCount(5).build();
     private static final Duration TIMEOUT = Duration.ofSeconds(30);
     private ServiceBuilder serviceBuilder;
     private TableStore tableStore;
-    private PravegaConnectionListener serverListener;
+    private PravegaConnectionListener server = null;
     private ConnectionFactory connectionFactory;
+    private TestingServer zkTestServer = null;
+    private ControllerWrapper controllerWrapper = null;
     private Controller controller;
     private KeyValueTableFactory keyValueTableFactory;
 
+
     @Before
     public void setup() throws Exception {
         super.setup();
+
+        final int controllerPort = TestUtils.getAvailableListenPort();
+        final String serviceHost = ENDPOINT;
+        final int servicePort = TestUtils.getAvailableListenPort();
+        final int containerCount = 4;
```

Comment:
This is a configuration. Move it to the `private static final` section at the top of this file and make sure you pass this value to the Segment Store as well. Otherwise if we change this at one point the test will stop working.

Reply:
Ok

Reply:
done